sw/device/tests/alert_handler_lpg_sleep_mode_pings.c

```diff
@@ -0,0 +1,425 @@
+  }
+
+  // Test phase #1: ping-timeout = 256
+  // No ping_timeout_fail alert shall be fired
+  // To ensure that the ping mechanism won't send spurious failure.
+  while (test_step_cnt < rnd_num_iterations) {
+    // Configure and lock the alerts
+    CHECK_DIF_OK(
+        dif_alert_handler_is_ping_timer_locked(&alert_handler, &is_locked));
+    if (!is_locked) {
+      alert_handler_config(/*ping_timeout=*/256);
+    }
+
+    // Power-on reset
+    if (pwrmgr_testutils_is_wakeup_reason(&pwrmgr, 0)) {
+      LOG_INFO("POR reset");
+      // Set  the AON timer to send a wakeup signal in ~150us.
+      // Go to deep sleep mode
+      aon_timer_testutils_wakeup_config(&aon_timer, 30);
+      enter_low_power(/*deep_sleep=*/(test_step_cnt % 2));
+    } else { /*wakeup reset*/
+      // Check the wakeup reason and do the cleanup
+      check_wakeup_reason();
+      cleanup_wakeup_src();
+
+      // After the wakeup, wait enough, and
+      // check that no ping-timeout-fail alert is fired
+      wait_enough_for_alert_ping();
+      CHECK_DIF_OK(dif_alert_handler_local_alert_is_cause(
+          &alert_handler, kDifAlertHandlerLocalAlertAlertPingFail, &is_cause));
+      CHECK(!is_cause,
+            "Expected response is ping_timeout_fail= 0! But we got "
+            "ping_timeout_fail = 1");
+
+      // Increment the test_step counter
+      flash_ctrl_testutils_counter_increment(&flash_ctrl, kCounterTestSteps);
+      test_step_cnt = flash_ctrl_testutils_counter_get(kCounterTestSteps);
+      // LOG_INFO("Test %d beginn", test_step_cnt);
+      // Set  the AON timer to send a wakeup signal in ~150us.
+      aon_timer_testutils_wakeup_config(&aon_timer, 30);
+      // Enter the normal sleep or deep sleep mode
+      enter_low_power(/*deep_sleep=*/(test_step_cnt % 2));
+      // enter_low_power(/*deep_sleep=*/rand_testutils_gen32_range(0,1));
+    }
+  }
+
+  // Test phase #2: ping_timeout = 2
+  // ping_timeout_fail alert shall be fired.
+  // To ensure that ping mechanism will continue to send out pings after wakeup
+  while (test_step_cnt < 2 * rnd_num_iterations) {
+    // Configure and lock the alerts
+    CHECK_DIF_OK(
+        dif_alert_handler_is_ping_timer_locked(&alert_handler, &is_locked));
+    if (!is_locked) {
+      alert_handler_config(/*ping_timeout=*/2);
+    }
+
+    if (pwrmgr_testutils_is_wakeup_reason(&pwrmgr, 0)) {
+      LOG_INFO("POR reset");
+      // Set  the AON timer to send a wakeup signal in ~150us.
+      // Go to sleep mode
+      aon_timer_testutils_wakeup_config(&aon_timer, 30);
+      enter_low_power(/*deep_sleep=*/(test_step_cnt % 2));
+    } else {
+      // Check the wakeup reason and do the cleanup
+      check_wakeup_reason();
+      cleanup_wakeup_src();
+
+      // After the wakeup, clear the alert_cause regs, wait enough, and
+      // check that ping-timeout-fail alert is fired
+      alert_handler_clear_cause_regs();
+      wait_enough_for_alert_ping();
```

Comment:
Hm at some point we need to wait to ensure there was enough time for a ping to be issued. So I think the `wait_enough_for_alert_ping();` should be OK here.

However, it probably makes more sense to move the `alert_handler_clear_cause_regs();` after the suggested alert check below to make sure we check for any spurious alerts first, and then clear before going through another sleep cycle.

We may have to update the test description accordingly (which is ok!).

Reply:
@msfschaffner, @cindychip thanks a lot for the comments.

In the new version of the code, I removed the `alert_handler_clear_cause_regs()` from the phase 1, so it can check for the spurious alerts.

In the phase 2, the purpose is to check if the alert handler ping mechanism keeps firing the alerts. Therefore, in phase 2, I now first clear the alerts and then check if there is new alerts coming.

Reply:
Thanks for changing that. LGTM